docs/notebooks/flax_basics.ipynb

```diff
@@ -0,0 +1,911 @@
+{
```

Comment:
<p>Do we need the <code>optimizer_def</code>  for anything? Why not directly </p><p><code>optimizer =&nbsp;optim.GradientDescent(learning_rate=alpha).create(init_params)</code></p><p> </p><p>Maybe <code>grad_fn</code>  -&gt; <code>loss_grad_fn</code>  (I would expect the <code>grad_fn</code>  to simply return the gradient)</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/google/flax/pull/542/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='16'/>

Reply:
<p>I don't know, this is a good question: the point of having two objects was to highlight the different concepts. Since those are different in the design of the API I thought it would be better to separate. Happy to consider other opinions.</p><p> </p><p>Agreed on the grad vs val + grad indeed!</p>